# Test notebook for boosting

The goal of this notebook is to test the algorithms on ```boosting.py```. We will teste them on four datasets:

- Diabetes dataset
- Prostate Cancer dataset
- California Housing Dataset
- Leukemia Dataset



## Diabetes Dataset

We can import this dataset:

In [31]:
from sklearn.datasets import load_diabetes
import numpy as np
import boosting

data = load_diabetes().data
response = load_diabetes().target

In [6]:
data.shape, response.shape

((442, 10), (442,))

In [10]:
data[0]

array([ 0.03807591,  0.05068012,  0.06169621,  0.02187239, -0.0442235 ,
       -0.03482076, -0.04340085, -0.00259226,  0.01990749, -0.01764613])

We have n = 442 and p = 10. Next we will normalice our data and our responses


In [28]:
def normalize_data(data,responses):
    y = responses - np.mean(responses)
    X = data - np.mean(data, axis=0)
    X = X / np.linalg.norm(X, axis=0)
    return X, y

In [29]:
X,y = normalize_data(data,response)

Know we are ready to apply our boosting algorithsm

In [ ]:
bossting = {}



